modules/core/src/main/java/org/apache/ignite/internal/processors/cache/IgniteCacheOffheapManager.java

```diff
@@ -670,7 +670,12 @@ interface CacheDataStore {
         long updateCounter();
 
         /**
-         * @return Reserved counter (HWM).
+         * @return Highest applied update counter (HWM).
```

Comment:
Let's discuss terms again. From my understanding, `highestAppliedCounter != HWM` but `reservedCounter == HWM`. My understanding is based on the Ignite design doc about data consistency [1]. 

Occasionally, we can say that a backup is aware of only `highestAppliedCounter` and then it can think about it as HWM. But backups don't send them to other nodes, then it just a useless info. From the other side primary nodes are aware of `reservedCounter` as the highest known counter value, and use it to send to other nodes. 

[1] https://cwiki.apache.org/confluence/display/IGNITE/Data+consistency

Reply:
ok. I'll add only the method. Without HWM docs renaming.